tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx

```diff
@@ -818,6 +1075,34 @@ describe('WidgetBuilder', function () {
         );
       });
     });
+
+    it('disables dashboards with max widgets', async function () {
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/dashboards/',
+        body: [
+          {...untitledDashboard, widgetDisplay: []},
+          {...testDashboard, widgetDisplay: [DisplayType.TABLE]},
+        ],
+      });
+
+      Object.defineProperty(dashboardsTypes, 'MAX_WIDGETS', {value: 1});
+
+      renderTestComponent({
+        query: {
+          source: DashboardWidgetSource.DISCOVERV2,
+        },
+      });
+
+      userEvent.click(await screen.findByText('Select a dashboard'));
+      userEvent.hover(screen.getByText('Test Dashboard'));
+      await waitFor(() => {
+        expect(
+          screen.getByText(
+            textWithMarkupMatcher('Max widgets (1) per dashboard reached.')
```

Comment:
This test was consistently failing in this PR. I noticed that most use-cases have `screen.getByText(textWithMarkupMatcher...)` so I converted this to a `waitFor` callback and it's passing now.

I also moved this up because this block deals with the "Add to Dashboard" flow and this field should only be available there.